Update README.md (#8)
bcgov/how-to-workshops: labs/netpol-quickstart/README.md

> Getting started with Kubernetes Network Policies In Openshift

Author: ```suggestion
## Getting started with Kubernetes Network Policies on Private Cloud Openshift Platform
```

> ## Getting started with Kubernetes Network Policies on Private Cloud Openshift Platform

Author: ```suggestion
---
title: Getting started with Kubernetes Network Policies on Private Cloud Openshift Platform
description: This resource explains the concept of the Zero-Trust security model implemented on the Private Cloud Openshift Platform using OPenshift 4 Built-In Software Definted Network (SDN) and provides guidance for creating network security rules for new projects to open up communications within their namespaces.
tags:
  - network security
  - software defined network
  - SDN
  - network policy
  - Kubernetes network policy
  - KNP
  - firewall
  - devops
  - private cloud
---

## Getting started with Kubernetes Network Policies on Private Cloud Openshift Platform
```

Read more about [the KNPs features](https://docs.openshift.com/container-platform/4.8/networking/network_policy/about-network-policy.html) supported in the current version of the OpenShift (v4.8) on the Platform. Note that the support for egress rules and ipBlock rules have only become available in OCP 4.8.

# Getting Started

Before we dive into the quick start policies, lets go over a few important details:

### Egress Rules

With the quick start KNP in place pods will be able to connect to other pods within their namespace, in other namespaces, or to external systems (outside of the cluster). 

Projects that require cross-namespace communication should reach out to the Platform Services Teams in Rocketchat ; these policies can be implemented, as needed, by a cluster administrator.

### Roll Out

As product teams implement network policy they are "rolling out" KNP; there is nothing Platform Services needs to do. Everything is in place and working as expected. 

> One a KNP is installed in every namespaces provisioned by the Platform Project Registry and it cannot be removed; if you remove it a smart robot will just re-create it a few moments later.

Author: ```suggestion
One KNP  is installed by default in every namespaces provisioned by the Platform Project Registry and it cannot be removed; if you remove it a smart robot will just re-create it a few moments later. This "deny-all" KNP enforces the Zero Trust model within the namespace and blocks all communications from and within the namespace. In order to enable specific connections, you will need to create your own custom KNPs that contain the rules that allow these communications.
```

# Prologue

>  Back in 2019 we decided to take a strong stance on security and, by way of a security focused project, began implementing several tools to make our OpenShift Container Platform (OCP) a leader in this respect. Openshift 4 Build-In Software Defined Network (SDN) has been implemented to control network security for all apps hosted on the Private Cloud Openshift Platform. 

Author: ```suggestion
 Back in 2019 we decided to take a strong stance on security and, by way of a security focused project, began implementing several tools to make our OpenShift Container Platform (OCP) a leader in this respect. Openshift 4 Built-In Software Defined Network (SDN) has been implemented to control network security for all apps hosted on the Private Cloud Openshift Platform. 
```

# Introduction

> Developers interact with the Openshift 4 Build-In SDN through the use of Kubernetes Network Policies (KNPs) objects that allow to specify network security rule declaratively within each namespace and down to the pod level. **All projects first provisioned on the Platform have "Zero-Trust" network security enabled by default** which means that all communications our and within the project namespaces are shutdown except for those that are explicitely allows in the KNPs.  This guide will walk you through implementing the quick start Network Policy (KNP) to unblock basic communications within a namespace. While this will be enough to your project up-and-running we **strongly** recommend rolling out more robust NPs to ensure your environment(s) are as secure as they can be. Further workshops will expand on this subject. 

Author: ```suggestion
Developers interact with the Openshift 4 Build-In SDN through the use of Kubernetes Network Policies (KNPs) objects that allow to specify network security rule declaratively within each namespace and down to the pod level. **All projects first provisioned on the Platform have "Zero-Trust" network security enabled by default** which means that all communications from and within the project namespaces are shutdown except for those that are explicitely allows in the KNPs.  This guide will walk you through implementing the quick start Network Policy (KNP) to unblock basic communications within a namespace. While this will be enough to your project up-and-running we **strongly** recommend rolling out more robust NPs to ensure your environment(s) are as secure as they can be. Further workshops will expand on this subject. 
```

Read more about [the KNPs features](https://docs.openshift.com/container-platform/4.8/networking/network_policy/about-network-policy.html) supported in the current version of the OpenShift (v4.8) on the Platform. Note that the support for egress rules and ipBlock rules have only become available in OCP 4.8.

# Getting Started

Before we dive into the quick start policies, lets go over a few important details:

### Egress Rules

With the quick start KNP in place pods will be able to connect to other pods within their namespace, in other namespaces, or to external systems (outside of the cluster). 

> Without egress policy, pods that need to communicate **between namespaces** only require ingress rules on the destination pod to permit the inbound communication. This is fine is most circumstances because you will have a *deny-by-default* rule guarding all your namespaces.

Author: ```suggestion
```

> High security projects that require egress rules to a specific namespace should reach out to Platform Services; these policies can be implemented, as needed, by a cluster administrator.

Author: ```suggestion
Projects that require cross-namespace communication should reach out to the Platform Services Teams in Rocketchat ; these policies can be implemented, as needed, by a cluster administrator.
```

> ### Roll Out

Author: ```suggestion
### Default Zero-Trust KNP
```
